tests/compute.py

```diff
@@ -359,3 +360,70 @@ def compute_image_load_store(cfg: SceneCfg, show_dbg_points=False):
         group.add_children(get_debug_points(cfg, cuepoints))
 
     return group
+
+
+_IMAGE_3D_STORE_COMPUTE = """
+void main()
+{
+    ivec2 pos = ivec2(gl_LocalInvocationID.xy);
+    float size = float(imageSize(texture).x);
+    float color = (float(pos.x) + float(pos.x * pos.y)) / (size * size) ;
+    imageStore(texture, ivec3(pos, 0), vec4(color));
+    imageStore(texture, ivec3(pos, 1), vec4(color));
+    imageStore(texture, ivec3(pos, 2), vec4(color));
+}
+"""
+
+_IMAGE_3D_LOAD_STORE_COMPUTE = """
+void main()
+{
+    ivec2 pos = ivec2(gl_LocalInvocationID.xy);
+    vec4 color;
+    color.r = imageLoad(texture, ivec3(pos, 0)).r;
+    color.g = imageLoad(texture, ivec3(pos, 1)).r;
+    color.b = imageLoad(texture, ivec3(pos, 2)).r;
+    color.a = 1.0;
+    color.rgb = (color.rgb * scale.factors.x) + scale.factors.y;
+    imageStore(texture_rgba, pos, color);
+}
+"""
+
+
+@test_cuepoints(points=_get_compute_histogram_cuepoints(), tolerance=1)
+@scene(show_dbg_points=scene.Bool())
+def compute_image_3d_load_store(cfg: SceneCfg, show_dbg_points=False):
+    size = _N
+    scale = ngl.Block(
+        fields=[ngl.UniformVec2(value=(-1.0, 1.0), label="factors")],
+        layout="std140",
+    )
+    texture = ngl.Texture3D(format="r32_sfloat", width=size, height=size, depth=3)
+    texture_rgba = ngl.Texture2D(width=size, height=size)
+    program_store = ngl.ComputeProgram(_IMAGE_3D_STORE_COMPUTE, workgroup_size=(size, size, 1))
+    program_store.update_properties(
+        texture=ngl.ResourceProps(as_image=True, writable=True),
+    )
+    compute_store = ngl.Compute(workgroup_count=(1, 1, 1), program=program_store)
+    compute_store.update_resources(
+        texture=texture,
+    )
```

Comment:
Do you mind adding a line break here so we can see more easily the 3 main passes (compute store, compute load & store, render load)?

Reply:
Done.